awx/main/scheduler/task_manager.py

```diff
@@ -623,6 +615,13 @@ def process_pending_tasks(self, pending_tasks):
                         tasks_to_update_job_explanation.append(task)
                 continue
 
+            if isinstance(task, WorkflowJob):
+                # Previously we were tracking allow_simultaneous blocking both here and in DependencyGraph.
+                # Double check that using just the DependencyGraph works for Workflows and Sliced Jobs.
+                self.dependency_graph.add_job(task)
```

Comment:
Not a critique of your PR, but we should consider moving the call to `self.dependency_graph.add_job` into the `start_task` method, as I cannot find a case where we call `start_task` without adding it to the dependency graph. I am aware that the `start_task` method does a lot of things, and it's extra mental energy to question whether this line in appropriate or not, when it should always be changing from pending to running.

Reply:
agree